test/Microsoft.Azure.Devices.Edge.Test.Common/linux/EdgeDaemon.cs

```diff
@@ -17,40 +17,51 @@ public class EdgeDaemon : IEdgeDaemon
 
         public static async Task<EdgeDaemon> CreateAsync(CancellationToken token)
         {
-            string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
-            if (platformInfo.Length == 1)
+            SupportedPackageExtension packageExtension;
+            string os, version;
+            string[] platformInfo = await Process.RunAsync("cat", @"/etc/os-release", token);
+            if (platformInfo.Any(s => s.Contains("Mariner")))
             {
-                platformInfo = platformInfo[0].Split(' ');
+                os = "Mariner";
+                version = "1.0";
+                packageExtension = SupportedPackageExtension.RpmMariner;
             }
-
-            string os = platformInfo[0].Trim();
-            string version = platformInfo[1].Trim();
-            SupportedPackageExtension packageExtension;
-
-            switch (os)
+            else
             {
-                case "Ubuntu":
-                    os = os.ToLower();
-                    packageExtension = SupportedPackageExtension.Deb;
-                    break;
-                case "Raspbian":
-                    os = "debian";
-                    version = "stretch";
-                    packageExtension = SupportedPackageExtension.Deb;
-                    break;
-                case "CentOS":
-                    os = os.ToLower();
-                    version = version.Split('.')[0];
-                    packageExtension = SupportedPackageExtension.Rpm;
+                platformInfo = await Process.RunAsync("lsb_release", "-sir", token);
```

Comment:
This works, but if we already ran `cat /etc/os-release`, do we still need to run `lsb_release -sir`? Doesn't the first command have everything we need?

Reply:
That is actually a great suggestion. From the /etc/os-release output, I would take the lines "ID=xxx" and "VERSION=yyy" and feed the contents into the "os" and "version" variables. @damonbarry , does that sound good to you?

Reply:
Sounds good, as long as we're confident `/etc/os-release` is available on all our tested platforms.

Reply:
It is, and this stackoverflow post gives me high confidence that it is a better solution in overall:
https://stackoverflow.com/questions/47838800/etc-lsb-release-vs-etc-os-release

Preparing and testing according changes.